### ISSUE TITLE
`MarkdownText` conflicts with `Modifier.clickable

### ISSUE THREAD

**Issue by jber18** [colintheshots/MarkdownTwain#7]
**Description:**  
When using the `MarkdownText` composable from the library, it renders the markdown content using an `AndroidView`. However, I encountered an issue where **adding a `Modifier.clickable` to the `MarkdownText` or any of its parent containers does not register click events** properly. This behavior creates conflicts when I try to wrap the `MarkdownText` with clickable elements for interaction.

---

### **Steps to Reproduce:**

1. Use `MarkdownText` in a Jetpack Compose component.
2. Try wrapping it inside a `Box` or `Row` with a `Modifier.clickable` to detect taps.
3. Observe that the click events are **not registered**, and the `onClick` callback is never triggered.

```kotlin
@Composable
fun TestMessage() {
    Box(
        modifier = Modifier
            .clickable { // This never gets called
                println("Clicked!")
            }
            .padding(16.dp)
    ) {
        MarkdownText(
            markdown = "This is **markdown** text!",
            modifier = Modifier.fillMaxWidth()
        )
    }
}
```

---

### **Expected Behavior:**

- The `Modifier.clickable` should register the click event when the markdown text is tapped.
- The `onClick` callback provided to the `Modifier.clickable` should be triggered.

---

### **Actual Behavior:**

- The `clickable` modifier does not work when applied to `MarkdownText` or its parent containers.
- It seems that the `AndroidView` used internally by `MarkdownText` is **blocking gesture events** from reaching the Jetpack Compose layout.

---

### **Possible Cause:**

The `MarkdownText` composable uses **`AndroidView`** under the hood to render markdown content via the `Markwon` library. Since `AndroidView` bridges the Android View system with Jetpack Compose, **it consumes touch events** internally, preventing them from propagating properly to the outer composables. 

This is a known issue when mixing **`AndroidView` with Compose gesture handlers**, as the Android view hierarchy does not seamlessly cooperate with the Compose UI system.

---

### **Workaround Attempts:**

I tried a few potential solutions, but none resolved the issue:
1. **Adding a clickable Modifier to the parent Box:** No success; the click is never registered.
2. **Adding a transparent overlay for gesture detection:** This blocked the visual content, which is not ideal for UI.
3. **Using zIndex adjustments:** Had no effect, as AndroidView still intercepts the gestures.

---

### **Suggested Solutions:**

1. **Expose a `onClick` callback** directly in the `MarkdownText` composable. This way, the click events can be handled internally within the `AndroidView`.
   
   Example API:
   ```kotlin
   @Composable
   fun MarkdownText(
       markdown: String,
       modifier: Modifier = Modifier,
       onClick: () -> Unit = {}
   ) { ... }
   ```

2. Alternatively, **provide guidance** on how to integrate `MarkdownText` with Compose's `clickable` modifier or gesture detection.

---

### **Environment:**

- Jetpack Compose version: `latest`
- Library version: `latest`
- Android Studio version: `latest`

---

**Comment by colintheshots** [comment]
I deliberately didn't want clicks to get intercepted at the top-level, so URLs would be clickable with their own handlers. One usually wants clicks to be handled at the lowest-level.

What level are you expecting to be clickable? The TextView? I assume you wouldn't want to also handle clicks on URLs or anything linkified in the same click handler?

**Comment by lneugebauer** [comment]
> What level are you expecting to be clickable? The TextView? I assume you wouldn't want to also handle clicks on URLs or anything linkified in the same click handler?

I'd like to add an `onLongClick` event to my `TextView`. Similar to Signal or WhatsApp, I want links to be clickable, but when the `TextView` is long clicked, a secondary action (In my case copying the text to the clipboard) should be triggered.

**Comment by lynnwilliam** [comment]
I agree, we need a way to have Clickable for those who want to add a long click, or any type of click for putting data on the clipboard.
also it the compose SelectionContainer{ markdown() } also doesn't work.
Please make the library more flexible instead of deciding how it should be used.

**Comment by miloszelko** [comment]
![Screenshot 2025-01-15 at 12 46 57](https://github.com/user-attachments/assets/dbc1b075-9bed-46f8-9e88-035b09de0e61)

@lneugebauer 
I had the same issue, here's workaround I used. Not the most elegant solution but it works.

**Comment by mitrejcevski** [comment]
@miloszelko how do you resolve `findViewById` inside the LaunchedEffect?

**Comment by colintheshots** [comment]
LocalContext.current keeps the local context inside of Compose and you can pass it in. Just be careful about passing context to long running jobs as you may not want memory leaks after the view should go away.
